libtenzir/src/execution_node.cpp

```diff
@@ -645,9 +662,17 @@ struct exec_node_state : inbound_state_mixin<Input>,
     }
   };
 
-  auto print_metrics() -> void {
-    const auto elapsed = std::chrono::duration_cast<duration>(
+  auto emit_metrics() -> void {
```

Comment:
We need to additionally call this function in the exit handler.

Reply:
Added an exit handler with last metrics emission in https://github.com/tenzir/tenzir/pull/3376/commits/aace5e16bbc0095d05eb231ad0716f1d7a1076cc

Reply:
That looks correct to me. Thanks!